File: Desktop/SummerVacation/0702/src/numTwentySeven.java
import java.util.Scanner;
public class numTwentySeven {
    public static void main(String[] args) {
        numTwentySeven check = new numTwentySeven();
        Scanner scan = new Scanner(System.in);
        String fiName,laName,code,id;
        System.out.print("Enter the first name: ");
        fiName = scan.next();
        System.out.print("Enter the last name: ");
        laName = scan.next();
        System.out.print("Enter the ZIP code: ");
        code = scan.next();
        System.out.print("Enter an employee ID: ");
        id = scan.next();
        check.validateInput(fiName,laName,code,id);
    }
    public boolean checkID(String id){
        if(id.length() == 7){//if length is perfect
            if(!Character.isLetter(id.charAt(0))&&!Character.isLetter(id.charAt(1)))// if first 2 letters isn't perfect
                return false;
            else if(id.charAt(2)!='-')// if 3rd letter isn't -
                return false;
            else{
                char code[] = id.toCharArray();
                for(int count = 3 ; count < id.length(); count++){//check other letters
                    if(!Character.isDigit(code[count]))//if other letters aren't numbers
                        return false;
                }
                return true;//everything is perfect
            }
        }
        else{// length is wrong
            return false;
        }
    }
    public boolean validateInput(String firstName, String secondName, String zipCode, String employeeId){
        boolean result = true;
        String message = null;
        if(firstName.length() < 2){
            message = firstName + " is too short\n";
            result = false;
        }
        if(secondName.length() < 2){
            message = message + secondName + " is too short.\n";
            result = false;
        }
        if(!checkID(employeeId)){
            message = message + employeeId + " is wrong.\n";
            result = false;
        }
        char[] code = zipCode.toCharArray();
        for(char c : code){
            if(!Character.isDigit(c)){
                result = false;
                message = message + zipCode + " is wrong.\n";
                System.out.print(message);
                return result;
            }
        }
        if(result){
            System.out.print("There are no errors.");
        }
        else{
            System.out.print(message);
        }
        return result;
    }
}

File: Desktop/SummerVacation/0702/python/numTwentySeven.py
def validateInput(fi_name,la_name,code,employee) :
    message = "\0"
    if len(fi_name) < 2 :
        message = message + fi_name + " is too short.\n"
    if len(la_name) < 2 :
        message = message + la_name + " is too short.\n"
    if not code.isdigit() :
        message = message + code + " isn't number.\n"
    if len(employee) == 7 :
        if not (employee[0].isalpha() and  employee[1].isalpha()) :
            message = message + employee + "'s first two letter will be letter.\n"
        if employee[2] != '-' :
            message = message + employee + "'s third letter will be -.\n"
        if not( employee[3].isdigit() and employee[4].isdigit() and employee[5].isdigit() and employee[6].isdigit()) :
            message = message + employee + "'s fourth to last letter will be number.\n"
    else :
        message = message + employee + "i s too short.\n"
    if message == "\0" :
        message = "There are no errors."
    return message
first_name = input("Enter the first name: ")
last_name = input("Enter the last name: ")
zip_code = input("Enter the zip code: ")
employee_id =input("Enter an employee ID: ")
result = validateInput(first_name,last_name,zip_code,employee_id)
print(result)